src/transformers/generation/utils.py

```diff
@@ -836,6 +836,9 @@ def _get_logits_processor(
         encoder_input_ids: torch.LongTensor,
         prefix_allowed_tokens_fn: Callable[[int, torch.Tensor], List[int]],
         logits_processor: Optional[LogitsProcessorList],
+        model_kwargs: Dict[str, Any],
+        negative_prompt_ids: Optional[torch.Tensor] = None,
+        negative_prompt_attention_mask: Optional[torch.Tensor] = None,
```

Comment:
Where does their docstring belong then?

Reply:
There is an assert checking for the absence of extra model_kwargs much earlier than the call to `_get_logits_processor()` that would consume them.
- Either we move this check later, and I'm not sure about the side effects
- Either we move those args back into GenerationConfig, consuming args _before_ the check
- Either they remain as separate args for `.generate()`

Reply:
@Vermeille @sanchit-gandhi I think we should leave them separate for now, as it is in the current state. Otherwise, it will be very hard for the users to discover how to use this feature. 

I'd suggest adding `This is an experimental feature, subject to breaking API changes in future versions.` in the docstring, in case we find a better design decision. (like we did [here](https://github.com/huggingface/transformers/blob/1e662f0f0716479b88fe4ea2c86fbbdec9ebc16c/src/transformers/models/llama/configuration_llama.py#L86))

Reply:
Not super keen about this design since it somewhat defeats the point of having the argument `model_kwargs` and opens the door for any number of arbitrary kwargs, but acknowledge the arguments for speed and discoverability, so will leave the final decision down to you for this PR